### PR TITLE
Issue/4823 show specific error message when total is less than 0.50 dollar

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Displays specific `amount_too_small` error in Stripe SDK v2

Closes: #4823 

### Description
This PR shows a specific and helpful error message when the user tries to collect payment less than $0.50. See [this documentation](https://stripe.com/docs/currencies#minimum-and-maximum-charge-amounts) from Stripe in order to know why the payment fails for the minimum amount for some currencies.

### Testing instructions

1. Go to the `Orders` screen
2. Select any product which has a total of **less than** `$0.50`
3. Click on the `Collect Payment` button from the order detail screen.
4. Make sure the reason for failure is appropriate, helpful, and makes sense.
5. Make sure the Primary action button in case of the `amount_too_small` error has a text `OK` and clicking on that just dismiss the dialog
6. Make sure the Primary action button in case of any other error has the text "Try again" and clicking on that re-initiates collecting payment
6. Run the unit tests and ensure those pass.

### Images/gif

https://user-images.githubusercontent.com/1331230/134857013-eaad3258-a419-4adb-bf52-bfa78524d737.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt [hunk 1]
```diff
@@ -449,7 +449,7 @@ class CardReaderPaymentViewModel
         data class FailedPaymentState(
             private val errorType: PaymentFlowError,
             override val amountWithCurrencyLabel: String?,
-            private val primaryLabel: Int? = R.string.try_again,
+            val primaryLabel: Int? = R.string.try_again,
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt]: Do we need this change?

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardPaymentStatus.kt [hunk 2]
```diff
@@ -15,13 +15,15 @@ sealed class CardPaymentStatus {
...
-        NO_NETWORK,
-        SERVER_ERROR,
-        PAYMENT_DECLINED,
-        GENERIC_ERROR,
-        AMOUNT_TOO_SMALL,
+    sealed class CardPaymentStatusErrorType {
+        object CardReadTimeOut : CardPaymentStatusErrorType()
+        object NoNetwork : CardPaymentStatusErrorType()
+        object ServerError : CardPaymentStatusErrorType()
+        sealed class PaymentDeclined : CardPaymentStatusErrorType() {
```
Comment (kidinov) [inline libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardPaymentStatus.kt:22]: 👍

**Review by kidinov — APPROVED** [review]
Great job! 👍

Just left one optional question

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt]: Good catch! Removed it by mistake while fixing merge conflicts. Fixed it here https://github.com/woocommerce/woocommerce-android/pull/4855/commits/91c7176c8b597324c8fff9fd8fd8abfb00c5a446

**Pull request merged**